app/src/main/java/com/pokemonshowdown/app/BattleFieldActivity.java

```diff
@@ -400,4 +436,104 @@ private void selectItem(int position) {
         mDrawerLayout.closeDrawer(mDrawerList);
     }
 
+    public void showErrorAlert(Exception e) {
+        AlertDialog alertDialog = createErrorAlert(e);
+        Log.e(BTAG, "App exception", e);
+        alertDialog.show();
+    }
+
+    public AlertDialog createErrorAlert(Exception e) {
+        return new AlertDialog.Builder(this)
+                .setMessage(e.toString())
+                .create();
+    }
+
+
+    public class DownloadUpdateTask extends AsyncTask<Void, Integer, Void> {
```

Comment:
This class is getting way too complicated. Can you put this into a separate class instead of a subclass plz. You can pass BattleFieldActivity into the constructor